eventuals/do-all.h

```diff
@@ -67,11 +95,68 @@ struct _DoAll {
                               StoppedException()));
                   if (counter_.fetch_sub(1) == 1) {
                     // You're the last eventual so call the continuation.
-                    k_.Start(std::move(values_));
+                    std::optional<std::exception_ptr> exception =
+                        GetExceptionIfExists();
+
+                    CHECK(exception);
+                    try {
+                      std::rethrow_exception(*exception);
+                    } catch (const StoppedException&) {
+                      k_.Stop();
+                    } catch (...) {
+                      k_.Fail(std::current_exception());
+                    }
                   }
```

Comment:
Why aren't we interrupting everyone else like we are for the failure case above?